src/asl/sample_agent.asl

```diff
@@ -4,18 +4,38 @@
 
 /* Initial goals */
 
-!score.
+// Start with the seeking goal where our objective is to find the ball OR hear from another player
+// that they are going for the ball
+!seeking.
 
 /* Plans */
++!seeking : beside(ball) & found(goal_r) <- kick(goal_r); !assist.
++!seeking : beside(ball) & not found(goal_r) <- find(goal_r); !score.
+// We found the ball, so tell all players that we want to go for it
++!seeking : found(ball) & not beside(ball) <- .send([sample_agent1,sample_agent2],tell,playerGoingForBall); !score.
+// Check if the message came from ourself or another player
++playerGoingForBall[source(A)] <- checkifself(A).
```

Comment:
In the jason mind explorer you can see where the percept came from, is there no easier was to do this?

Reply:
The only other way I can think of doing it in AgentSpeak is to call a function from the environment before anything else happens, and save it's own name using a percept. Then when it wants to send the message it can take its name out from the list. But I think this way works well, so I'd keep it.